File: bench/go/mandel.go
package main

import "fmt"

const (
	W        = 4096
	H        = 4096
	MAX_ITER = 200
)

func mandel(cr, ci float64) int32 {
	var zr, zi float64 = 0.0, 0.0
	for i := int32(0); i < MAX_ITER; i++ {
		zr2 := zr * zr
		zi2 := zi * zi
		if zr2+zi2 > 4.0 {
			return i
		}
		newZr := zr2 - zi2 + cr
		zi = 2.0*zr*zi + ci
		zr = newZr
	}
	return MAX_ITER
}

func main() {
	var total int32 = 0
	for y := int32(0); y < H; y++ {
		ci := (float64(y)/float64(H))*2.0 - 1.0
		for x := int32(0); x < W; x++ {
			cr := (float64(x)/float64(W))*3.0 - 2.0
			total += mandel(cr, ci)
		}
	}
	fmt.Printf("mandel total = %d\n", total)
}

File: bench/c/mandel.c
#include <stdio.h>

#define W 4096
#define H 4096
#define MAX_ITER 200

static int mandel(double cr, double ci) {
    double zr = 0.0, zi = 0.0;
    for (int i = 0; i < MAX_ITER; i++) {
        double zr2 = zr * zr;
        double zi2 = zi * zi;
        if (zr2 + zi2 > 4.0) return i;
        double new_zr = zr2 - zi2 + cr;
        zi = 2.0 * zr * zi + ci;
        zr = new_zr;
    }
    return MAX_ITER;
}

int main(void) {
    int total = 0;
    for (int y = 0; y < H; y++) {
        double ci = ((double)y / (double)H) * 2.0 - 1.0;
        for (int x = 0; x < W; x++) {
            double cr = ((double)x / (double)W) * 3.0 - 2.0;
            total += mandel(cr, ci);
        }
    }
    printf("mandel total = %d\n", total);
    return 0;
}
